core/templates/dev/head/pages/exploration_editor/simple_editor_tab/ScrollSyncService.js

```diff
@@ -0,0 +1,110 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service that syncs the position of the user in the simple
+ * editor with the navigation sidebar, see ScrollSyncDirective.js.
+ */
+
+oppia.factory('ScrollSyncService', [
+  '$anchorScroll', '$document', '$rootScope', 'QuestionHashService',
+  'SimpleEditorManagerService',
+  function($anchorScroll, $document, $rootScope, QuestionHashService,
+           SimpleEditorManagerService) {
+    var targets = [];
+    var questionList = SimpleEditorManagerService.getQuestionList();
+
+    var clearHighlighted = function() {
+      angular.element('.highlighted').removeClass('highlighted');
+    };
+
+    var collapse = function(hash) {
+      var questions = questionList.getQuestions();
+      for (var i = 0; i < questions.length; i++) {
+        if (QuestionHashService.getQuestionHash(questions[i]) === hash) {
+          $rootScope.$broadcast(
+            'SimpleEditorSidebarToggleCollapse', questions[i]
+          );
+          break;
+        }
+      };
+    };
+
+    $anchorScroll.yOffset = 80;
+
+    var scrollSyncService = {
+      // Add an element to the targets list, when a user scrolls to an element
+      // in the target list, the sidebar is updated accordingly.
+      addTarget: function(target) {
+        targets.push(target);
+      },
+      // Update the active nav link
+      updateActive: function(hash) {
+        if (!angular.isDefined(hash)) {
+          return;
+        } else {
+          clearHighlighted();
+          var elm = angular.element('#sidebaritem-' + hash);
+
+          if (!(hash === 'intro' || hash === 'title')) {
```

Comment:
Ah I may have been unclear in my earlier comment. I think the highlight should appear on these things too. In general there should always be a highlight in the sidebar (otherwise it's hard for me to tell where I am).